### ISSUE TITLE
Inaccessible/inescapable UI when restricted to keyboard navigation

### ISSUE THREAD

**Issue by Seneral** [ocornut/imgui#7688]
### Version/Branch of Dear ImGui:

Version 1.90.8, Branch: master/docking

### Back-ends:

imgui_impl_glfw.cpp + imgui_impl_opengl3.cpp

### Compiler, OS:

Linux + Clang 17

### Details:

While trying to only use my app with keyboard navigation, I found some limitations where I had to fall back to using a mouse.

For example, I couldn't find a way to close a window that did not have an explicit close button within the UI - the X itself was not selectable. This became especially severe with modal windows that block Ctrl-Tab switching, meaning you cannot escape them once opened without a mouse.
For windows, maybe there needs to be a special way to close them (maybe from Ctrl-Tab switcher?). If there is a way already, I'll have to communicate that to my users once they start using keyboard navigation, because as a user it was not apparent to me. But for the modal window that has an opened property (and thus shows an X), pressing Escape should really close it.

In addition, accessing menu bars is not possible unless it's the main menu bar, in which case Ctrl+Tab will show it.
This applies to Menu Bars in windows, e.g. the Dear ImGui Demo window toolbar, and much less relevant, the menu bars in both the "Popups/With a menu.." popup and the "Modals/Stacked modals" modal.

Other than that, I am very impressed how well it works. I'm able to fully use my app with just the keyboard. Thank you!

**Comment by ocornut** [comment]
Have you tried tapping Alt to switch to the menu layer? Both of your problems should be solved by this :)

It’s the standard input in Windows land to do it. If you didn’t discover it, i am curious as to what input you would normally use on your system to do that?

**Comment by Seneral** [comment]
You're right, that works, my bad. Should've looked up basic controls. As you might have guessed, I don't normally use just the keyboard, I was testing the feature. Just talked with a UI Engineer about Dear ImGui and its use for desktop applications like mine, and was curious how well supported more niche use cases are (was playing with the idea of eventually integrating screen readers, this makes it a lot more realistic).

I'll definitely have to tweak my application a bit to make it work well. E.g. I have a log section with each entry being a selectable and a follow state for new logs - switching between that follow state and navigating logs is turning out to be quite annoying. More research on my part needed as to what how other applications handle such an interaction.

Thanks for your help, and sorry for not researching this properly.

**Comment by ocornut** [comment]
I double-checked and it is mentioned in the `ShowUserGuide()`:

![image](https://github.com/ocornut/imgui/assets/8225057/1b1f12c6-9a94-41f5-9aa1-c981067e04b9)

Also linking to #4122 about screen readers. I don't know much about them, and I think it may be difficult to meet all standards, but I am certain we could implement some base line features such as describing newly focused item while tabbing etc. We should need to be vary we are likely to be hitting some walls as dear imgui tends to not have enough holistic understanding of the UI setup, and some items may be lacking semantic, or dissociating focused item/identifier from e.g. labels. Typically when  you use a row-spanning Selectable() which is the item being interacted, it's a Selectable with no label, with some description separately.

I plan at some point to rework the "capturing" LogXXX api into a nicer API and we can probably push improvements there.
We also had an experimental feature (not currently pushed) in imgui_test_engine that extracted text and colors directly from vertex data. by reverse mapping UV coordinates to characters, which has its use for testing but has the limitation that any clipped text won't be seen by it (in a way, it would "see" the same thing as a user, but reading a clipper text by clipping the actual prononciation is not going to feel great..).
